src/classes/RD2_ValidationService_TEST.cls

```diff
@@ -200,6 +212,208 @@ private with sharing class RD2_ValidationService_TEST {
+    @isTest
+    private static void shouldNotCreateActiveRDWhenAmountIsNotValid() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Contact con = getContact();
+
+        TEST_RecurringDonationBuilder rdBuilder = new TEST_RecurringDonationBuilder()
+            .withContact(con.Id)
+            .withDefaultValues()
+            .withDayOfMonth('15');
+
+        for (Decimal amount : new Set<Decimal>{
+            null,
+            0,
+            -10
+        }) {
+            npe03__Recurring_Donation__c rd = rdBuilder
+                .withAmount(amount)
+                .build();
+
+            String errMessage = '';
+            try {
+                insert rd;
+            } catch (Exception e) {
+                errMessage = e.getMessage();
+            }
+            System.assert(errMessage.contains(System.Label.RD2_AmountMustBeValid),
+                'Amount validation error should be expected: ' + errMessage);
+        }
+    }
+
+    /***
+    * @description Verifies if it creates Recurring donations for closed status when
+        invalid amount is specified.
+    */
+    @isTest
+    private static void shouldCreateRDForClosedRDWhenAmountIsNotValid() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Contact con = getContact();
+
+        TEST_RecurringDonationBuilder rdBuilder = new TEST_RecurringDonationBuilder()
+            .withContact(con.Id)
+            .withDefaultValues()
+            .withDayOfMonth('15')
+            .withStatusClosed();
+
+        for (Decimal amount : new Set<Decimal>{
+            null,
+            0,
+            -10
+        }) {
+            npe03__Recurring_Donation__c rd = rdBuilder
+                .withAmount(amount)
+                .build();
+
+            String errMessage = '';
+            try {
+                insert rd;
+            } catch (Exception e) {
+                errMessage = e.getMessage();
+            }
+            System.assert(String.isBlank(errMessage), 'Recurring Donation should be created successfully: ' + errMessage);
+        }
+    }
+
+    /***
+    * @description Verifies if it displays error message when installment period
+        is monthly and day of month is blank
+    */
+    @isTest
+    private static void shouldFailWhenDayOfMonthIsBlankForMonthlyInstallmentPeriod() {
+        String errMessage = '';
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Contact con = UTIL_UnitTestData_TEST.getContact();
+        insert con;
+
+        npe03__Recurring_Donation__c rd = new TEST_RecurringDonationBuilder()
+            .withContact(con.Id)
+            .withAmount(10)
+            .withInstallmentPeriodMonthly()
+            .build();
+
+        try {
+            insert rd;
+        } catch (Exception e) {
+            errMessage = e.getMessage();
+        }
+
+        System.assert(errMessage.contains(System.Label.RD2_DayOfMonthIsRequiredForMonthlyInstallment));
```

Comment:
Missing expected result message, for example 
```
System.assert(
    errMessage.contains(System.Label.RD2_DayOfMonthIsRequiredForMonthlyInstallment),
    'Should indicate "Day Of Month" cannot be empty: ' + errMessage
);
```

Reply:
@RadSFDeveloper , Do you want to replace with assertEquals method?

Reply:
If possible, however the comment was with regards to message displayed if the assertion fails. For example, if the above test fails, the message will be "Assertion failed". If the assert statement is written as suggested, we would see the actual error message value. For example, if the actual message is "foo", the message displayed when the test fails will include it, ie "Assertion failed : Should indicate "Day Of Month" cannot be empty:  foo".